src/value/mod.rs

```diff
@@ -513,21 +557,18 @@ impl Hash for Value {
         match *self {
             Value::Null => 0.hash(state),
             Value::Bool(b) => (1, b).hash(state),
-            Value::I64(i) => (2, i).hash(state),
-            Value::F64(_) => {
-                // you should feel bad for using f64 as a map key
-                3.hash(state);
-            }
-            Value::String(ref s) => (4, s).hash(state),
-            Value::Sequence(ref seq) => (5, seq).hash(state),
-            Value::Mapping(ref map) => (6, map).hash(state),
+            Value::Number(ref i) => (2, i).hash(state),
+            Value::String(ref s) => (3, s).hash(state),
+            Value::Sequence(ref seq) => (4, seq).hash(state),
+            Value::Mapping(ref map) => (5, map).hash(state),
         }
     }
 }
 
 mod index;
 mod partial_eq;
 mod from;
+mod number;
 
 mod ser;
-mod de;
+mod de;
```

Comment:
Please set your editor to end every line of characters with a newline character. Otherwise this does not conform to the POSIX definition of a "text file" and will break some tools.